wgpu-types/src/lib.rs

```diff
@@ -448,6 +449,12 @@ impl fmt::Display for RequestAdapterError {
     }
 }
 
+impl error::WebGpuError for RequestAdapterError {
+    fn webgpu_error_type(&self) -> error::ErrorType {
+        error::ErrorType::Validation
+    }
+}
+
```

Comment:
I'm honestly not sure that we should implement this for `RequestAdapterError`, given that not finding an adapter is either expected (we might not _have_ an adapter to give, return `null` in WebGPU's JS parlance) or due to a configuration problem (which standard user agents shouldn't be exposing). WebGPU error classification doesn't actually apply here.

Reply:
I agree, same for `RequestDeviceError`.

Reply:
Implemented the removal from `Request{Device,Adapter}Error` as a separate commit on top: 6285e551.